docs/Specifications/Baseline CLI/baseline-workgroup/baseline-workgroup-setup.md

```diff
@@ -0,0 +1,134 @@
+
+The purpose of this command is to output `workgroup-config.json`. The purpose of the resulting artifact is to be supplied as an argument to the deployment command of the `baseline` CLI.
+
+### Format of the resulting artifact
+
+```json
+{
+	"contracts": {
+		"resolver": {
+			"type": string,
+			"value": string,
+			"extra": string
+		}
+	}
+}
+```
+
+The main purpose of this artifact is to specify the resolver for the workgroup configuration. In the initial version of it would only need to specify the resolver for the contract artifacts, but can be extended with further configurations with the progress of baseline.
+
+The `type` property of this resolver config will have several predefined values. In the first version these will be `path` - resolving to the `ContractArtifactsPathResolver` and `custom` for custom resolvers.
+
+Usage of each of those would be specified in the documentation and as an example at the top of the file will be provided.
+
+More info about the contract resolvers can be found in the "Interfaces and schemas for the main logical input objects and resolvers" specification.
+
+## baseline workgroup setup:organisation
+
+The syntax of this command would be `baseline workgroup setup:organisation [organisationName]`.
+
+The `organisationName` parameter would specify the name of the organisation being setup.
+
+The purpose of this command is to output `${organisationName}-organisation-config.json`. The purpose of the resulting artifact is to be supplied as an argument to the organisation registration command of the `baseline` CLI.
+
+### Purpose of the resulting artifact
+
+```json
+{
+	"organisationName": string,
+	"organisation": {
+		"resolver": {
+			"type": string,
+			"value": string, 
+			"extra": string
+		}
+	}
+}
+```
+
+The main purpose of this artifact is to specify the resolver for the organisation configuration.
+
+The `type` property of this resolver config will have several predefined values. In the first version these will be `path` - resolving to the `OrganisationConfigPathResolver` , `raw` - allowing these to be passed through the `value` parameter as string that can be parsed, and `custom` for custom resolvers.
+
+Usage of each of those would be specified in the documentation and an example will be provided at the top of the file.
+
+More info about the organisation resolvers can be found in the "Interfaces and schemas for the main logical input objects and resolvers" specification.
+
+## baseline workgroup setup:workflow
+
+The syntax of this command would be `baseline workgroup setup:workflow [workflowName]`.
+
+The `circuitName` parameter would specify the name of the circuit being setup.
+
+The purpose of this command is to output `${workflowName}-config.json`. The purpose of the resulting artifact is to be supplied as an argument to the workflow registration command of the `baseline` CLI.
+
+### Purpose of the resulting artifact
+
+```json
+{
+	"workflowName": string,
```

Comment:
consider `init-core` branch, which refers to a generic interface for different service interactions. I have been looking into it particularly to create a generic circuit component for a workflow. Possibly, these would provide more inputs to the documentation here. @kthomas @bitwiseguy 
https://github.com/ethereum-oasis/baseline/tree/init-core/baseline/core